test/macro_benchmark/example/memcpy_macro_benchmark.cpp

```diff
@@ -0,0 +1,32 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <algorithm>
+#include <cstring>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/test/performance/units.hpp>
+
+static void memcpy_benchmark(benchmark::State & state) {
+    size_t size = state.range(0);
+    char* src = new char[size];
```

Comment:
Dynamic allocated memory with 1 GB size? I thought std::array is compile time size on the stack.

Reply:
ah sorry, I was blended by the array notation and overlooked that you are doing dynamic allocations. Then again why not std::vector?

Reply:
And then of course using `src.data()` to get the pointer.

Reply:
Our other benchmark example is exactly the same as this one, but with a different size. (That is on the orher side a standard example in the google benchmark repository). I didn't want to reinvent everything and keep this file as independent as possible.

https://github.com/seqan/seqan3/blob/aa45291d7aafd88d2d58c2e4fcf1702b1fbf6220/test/performance/example/example_benchmark.cpp#L21-L33

BTW I recently watched this talk https://www.youtube.com/watch?v=GPP64opjy_Y and this is called différance.

Reply:
>  I didn't want to reinvent everything and keep this file as independent as possible.

Well, it is not about reinventing everything. And even if that is a standard example on google benchmark, we don't do C and I don't see why we should not stick to C++, especially on take-from-examples. I mean if at least there would be a comparison to copy using the vector interfaces, then this would make sense. But this example doesn't feel right to me:

```cpp
     unsigned size = state.range(0); 
     std::vector src(size, '-');
     std::vector dst(size);
  
     for (auto _ : state) 
         std::copy(src.begin(), src.end(), dst.begin());
  
     state.counters["bytes_per_second"] = seqan3::test::bytes_per_second(size); 
```